gdx/src/com/badlogic/gdx/utils/ObjectMap.java

```diff
@@ -1,846 +1,782 @@
+	 * every value, which may be an expensive operation.
+	 *
+	 * @param identity If true, uses == to compare the specified value with values in the map. If false, uses
+	 *                 {@link #equals(Object)}.
+	 */
+	public K findKey (Object value, boolean identity) {
+		V[] valueTable = this.valueTable;
+		if (value == null) {
+			K[] keyTable = this.keyTable;
+			for (int i = valueTable.length; i-- > 0; )
+				if (keyTable[i] != null && valueTable[i] == null)
+					return keyTable[i];
+		} else if (identity) {
+			for (int i = valueTable.length; i-- > 0; )
+				if (valueTable[i] == value)
+					return keyTable[i];
+		} else {
+			for (int i = valueTable.length; i-- > 0; )
+				if (value.equals(valueTable[i]))
+					return keyTable[i];
+		}
+		return null;
+	}
+
+	/**
+	 * Increases the size of the backing array to accommodate the specified number of additional items. Useful before adding many
+	 * items to avoid multiple backing array resizes.
+	 */
+	public void ensureCapacity (int additionalCapacity) {
+		int sizeNeeded = size + additionalCapacity;
+		if (sizeNeeded >= threshold)
+			resize(MathUtils.nextPowerOfTwo((int)Math.ceil(sizeNeeded / loadFactor)));
+	}
+
+	final void resize (int newSize) {
+		int oldCapacity = keyTable.length;
+		threshold = (int)(newSize * loadFactor);
+		mask = newSize - 1;
+		shift = Long.numberOfLeadingZeros(mask);
+
+		K[] oldKeyTable = keyTable;
+		V[] oldValueTable = valueTable;
+
+		keyTable = (K[])new Object[newSize];
+		valueTable = (V[])new Object[newSize];
+
+		int oldSize = size;
+		size = 0;
+		if (oldSize > 0) {
+			for (int i = 0; i < oldCapacity; i++) {
+				K key = oldKeyTable[i];
+				if (key != null)
+					putResize(key, oldValueTable[i]);
+			}
+		}
+	}
+
+	public int hashCode () {
+		int h = size;
+		K[] keyTable = this.keyTable;
+		V[] valueTable = this.valueTable;
+		for (int i = 0, n = keyTable.length; i < n; i++) {
+			K key = keyTable[i];
+			if (key != null) {
+				h ^= key.hashCode();
+
+				V value = valueTable[i];
+				if (value != null) {
+					h += value.hashCode();
+				}
+			}
+		}
+		return h;
+	}
+
+	public boolean equals (Object obj) {
+		if (obj == this)
+			return true;
+		if (!(obj instanceof ObjectMap))
+			return false;
+		ObjectMap other = (ObjectMap)obj;
+		if (other.size != size)
+			return false;
+		K[] keyTable = this.keyTable;
+		V[] valueTable = this.valueTable;
+		for (int i = 0, n = keyTable.length; i < n; i++) {
+			K key = keyTable[i];
+			if (key != null) {
+				V value = valueTable[i];
+				if (value == null) {
+					if (other.get(key, dummy) != null)
+						return false;
+				} else {
+					if (!value.equals(other.get(key)))
+						return false;
+				}
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Uses == for comparison of each value.
+	 */
+	public boolean equalsIdentity (Object obj) {
+		if (obj == this)
+			return true;
+		if (!(obj instanceof IdentityMap))
```

Comment:
This (and below) should be ObjectMap, not IdentityMap. Blame shows this was my fault! Probably copy/paste from IdentityMap and no one ever noticed.

Reply:
Fixing now. The only other places I see `equalsIdentity()` in this PR are IntMap and LongMap, and neither of those mentions IdentityMap.